url04f7fcf4e1_peteratt_HDFS_Replication_Simulator_tgz-pJ8-hdfs_replicationsimulator_MainJ8/src/hdfs/replicationsimulator/SimFileGenerator.java

```diff
@@ -0,0 +1,200 @@
+package hdfs.replicationsimulator;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class SimFileGenerator {
+
+	// Types:
+	// 0 is for constant
+	// 1 is for exponential
+	// 2 is for gaussian
+	private static final int CONSTANT = 0;
+	private static final int EXPONENTIAL = 1;
+	private static final int GAUSSIAN = 2;
+
+	private static String filename;
+
+	// Defaults
+	private static int distrType = 0;
+	private static int mttf = 30000;
+
+	private static int time;
+	private static int nodes;
+	private static int failingNodes;
+	private static int nBlocks;
+
+	private static List<Integer> failedNodes;
+
+	public static void main(String[] args) {
+		try {
+			parseArgs(args);
+		} catch (Exception e) {
+			System.err.println("Error while parsing parameters");
+			System.exit(1);
+		}
+	}
+
+	private static void parseArgs(String[] args) throws Exception {
+
+		for (String s : args) {
+			if (s.startsWith("file=")) {
+				filename = s.split("=")[1];
+				System.out.println("File is: " + filename);
+			}
+		}
+
+		File f = new File(filename);
+		FileWriter fw = new FileWriter(f);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced at lines 55, 58, 60, 62, 64, 70, 72, 87, 88. The resource is closed at line 92. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).